Mathlib/LinearAlgebra/TensorProduct/Basis.lean

```diff
@@ -66,5 +67,181 @@ lemma Basis.baseChange_apply (b : Basis ι R M) (i) :
     b.baseChange S i = 1 ⊗ₜ b i := by
   simp [Basis.baseChange, Basis.tensorProduct]
 
+variable (K : Type*) [CommSemiring K] (B C: Type*) [AddCommMonoid B] [Module K B]
+    [AddCommMonoid C] [Module K C]
+
+/-- Elements in B ⊗ C can be represented by sum of elements in B tensor elements of basis of C.-/
+lemma TensorProduct.eq_repr_basis_right
+    {ιC ιB : Type*} (𝒞 : Basis ιC K C) (ℬ : Basis ιB K B) (x : B ⊗[K] C) :
+    ∃ (s : Finset ιC) (b : ιC → B), ∑ i ∈ s, b i ⊗ₜ[K] 𝒞 i = x := by
```

Comment:
This statement looks quite weird. The assumption that B has a basis is not used in the statement and is not necessary in the proof; indeed all that one needs is that iC spans C as a K-vector space. Then one can write x=sum_i b_i (x) c_i, one writes each c_i in terms of the spanning set and tidies up. No hypotheses on B are necessary at all. It also feels like the most natural statement would be with a finsupp ιC →_0 B.

Reply:
Ah okay but I thought we need both basis for Basis.tensorproduct?

Reply:
In `TensorProduct.eq_repr_basis_right`, I used another proof avoiding `Basis.tensorProduct`; however in `TensorProduct.sum_tmul_basis_right_eq_zero`, I have to assume `M` is Free so tha tI can use `Basis.tensorProduct`